src/modules/reporting/components/reporting-report/reporting-report.jsx

```diff
@@ -0,0 +1,310 @@
+    this.validateOutcome = this.validateOutcome.bind(this)
+    this.validateNumber = this.validateNumber.bind(this)
+    this.validateStake = this.validateStake.bind(this)
+  }
+
+  prevPage() {
+    const newStep = this.state.currentStep <= 0 ? 0 : this.state.currentStep - 1
+    this.setState({ currentStep: newStep })
+  }
+
+  nextPage() {
+    const newStep = this.state.currentStep >= 1 ? 1 : this.state.currentStep + 1
+    this.setState({ currentStep: newStep })
+  }
+
+  validateIsMarketValid(isMarketValid) {
+    const updatedValidations = { ...this.state.validations }
+    updatedValidations.isMarketValid = true
+
+    if (isMarketValid) {
+      if (!updatedValidations.hasOwnProperty('selectedOutcome')) {
+        updatedValidations.selectedOutcome = false
+      }
+    } else if (updatedValidations.hasOwnProperty('selectedOutcome') && updatedValidations.selectedOutcome === false) {
+      delete updatedValidations.selectedOutcome
+    }
+
+    this.setState({
+      validations: updatedValidations,
+      isMarketValid
+    })
+  }
+
+  validateOutcome(selectedOutcome) {
+    const updatedValidations = { ...this.state.validations }
+    updatedValidations.selectedOutcome = true
+
+    this.setState({
+      validations: updatedValidations,
+      selectedOutcome
+    })
+  }
+
+  validateNumber(fieldName, value, humanName, min, max) {
+    const updatedValidations = { ...this.state.validations }
+
+    const minValue = parseFloat(min)
+    const maxValue = parseFloat(max)
+
+    switch (true) {
+      case value === '':
+        updatedValidations[fieldName] = `The ${humanName} field is required.`
+        break
+      case (value > maxValue || value < minValue):
+        updatedValidations[fieldName] = `Please enter a ${humanName} between ${min} and ${max}.`
+        break
+      default:
+        updatedValidations[fieldName] = true
+        break
+    }
+
+    this.setState({
+      validations: updatedValidations,
+      [fieldName]: value
+    })
+  }
+
+  validateStake(stake) {
+    const updatedValidations = { ...this.state.validations }
+
+    switch (true) {
+      case stake === '':
+        updatedValidations.stake = `The stake field is required.`
+        break
+      case stake <= 0:
+        updatedValidations.stake = `Please enter a stake greater than 0.`
+        break
+      default:
+        updatedValidations.stake = true
+        break
+    }
+
+    this.setState({
+      validations: updatedValidations,
+      stake
+    })
+  }
+
+  render() {
+    const s = this.state
+    const p = this.props
+
+    return (
+      <section>
+        <Helmet>
+          <title>Submit Report</title>
+        </Helmet>
+        <MarketPreview
+          {...p.market}
+          isLogged={p.isLogged}
+          location={p.location}
+          history={p.history}
+          cardStyle="single-card"
+          buttonText="View"
+        />
+        <article className={FormStyles.Form}>
+          { s.currentStep === 0 &&
```

Comment:
Go ahead and split the steps into separate components — just looking at what’s there, they could simply be pure.